### PR TITLE
ClusterLoader - Increasing timeout for waiting for deletion operations

### PR DESCRIPTION
**Pull request by krzysied**
Increasing timeout for operation related to the pod deletion.

This change is required due to the garbage collector requiring twice as much time to remove the pods as it was needed for pods creation.

### CODE REVIEW

File: clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go [hunk 1]
```diff
@@ -304,16 +304,41 @@ func (w *waitForControlledPodsRunningMeasurement) handleObject(clients *framewor
 	if err := w.deleteObjectLocked(clients.GetClient(), oldRuntimeObj); err != nil {
 		glog.Errorf("%s: delete checker error: %v", w, err)
 	} else if newRuntimeObj == nil {
-		if err := w.handleObjectLocked(clients.GetClient(), oldRuntimeObj, true); err != nil {
+		if err := w.handleObjectLocked(clients.GetClient(), oldRuntimeObj, true, true); err != nil {
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go]: I'm not a fan of passing a list of bools, as this is hard to follow.

I think it would be clear if handleObjectLocked would take both old and new version and encapsulate that logic in the handleObjectLock function?

Comment (krzysied) [reply clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go]: Done.

Comment (krzysied) [reply clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go]: As you suggested, I moved most of the logic to the handleObjectLocked function.

File: clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go [hunk 2]
```diff
@@ -303,29 +303,63 @@ func (w *waitForControlledPodsRunningMeasurement) handleObject(clients *framewor
...
+	newReplicas, err := runtimeobjects.GetReplicasFromRuntimeObject(newObj)
+	if err != nil {
+		return false, err
+	}
+
+	return newReplicas < oldReplicas, nil
+}
+
+func (w *waitForControlledPodsRunningMeasurement) handleObjectLocked(c clientset.Interface, oldObj, newObj runtime.Object) error {
+	if oldObj == nil && newObj == nil {
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go]: This should never be the case.

File: clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go [hunk 3]
```diff
@@ -303,29 +303,63 @@ func (w *waitForControlledPodsRunningMeasurement) handleObject(clients *framewor
...
 
-func (w *waitForControlledPodsRunningMeasurement) handleObjectLocked(c clientset.Interface, obj runtime.Object, isDeleted bool) error {
-	if obj == nil {
+func checkScaledown(oldObj, newObj runtime.Object) (bool, error) {
+	if newObj == nil {
+		return true, nil
+	}
+	if oldObj == nil {
+		return false, nil
+	}
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go]: Instead of having those if`s, can you please:
- add "if obj == nil return 0" statement to GetReplicasFromRuntimeObject function (similarly as it is handled in GetSpecFrom..
- remove those if`s here

File: clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go (hunk 2 above)
Comment (krzysied) [reply clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go]: Done.

File: clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go (hunk 3 above)
Comment (krzysied) [reply clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go]: Done.
